benchmark/benchmark.go

```diff
@@ -30,3 +36,101 @@ func GetSubject() (string, error) {
 	}
 	return subject, nil
 }
+
+// Wrapper reports the benchmarking metrics of targetFunc to testing.B.
+func Wrapper(b *testing.B, targetFunc func(), cleanupFunc func()) {
+	var peakCPUSum, avgCPUSum float64
+	var totalCPUTimeSum time.Duration
+	var diskUsageDeltaSum int64
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		b.StartTimer()
+		peakCPU, avgCPU, totalCPUTime, diskUsageDelta, err := measureMetrics(targetFunc)
```

Comment:
Should we create a struct and return the struct instead of multiple variables?
eg.
```
mesuredMatrics, err := easureMetrics(targetFunc);
```